prow/plank/controller.go

```diff
@@ -267,27 +271,10 @@ func (c *Controller) SyncMetrics() {
 // in-place when it aborts.
 // TODO: Dry this out - need to ensure we can abstract children cancellation first.
 func (c *Controller) terminateDupes(pjs []prowapi.ProwJob, pm map[string]coreapi.Pod) error {
-	// "job org/repo#number" -> newest job
-	dupes := make(map[string]int)
-	for i, pj := range pjs {
-		if pj.Complete() || pj.Spec.Type != prowapi.PresubmitJob {
-			continue
-		}
-		n := fmt.Sprintf("%s %s/%s#%d", pj.Spec.Job, pj.Spec.Refs.Org, pj.Spec.Refs.Repo, pj.Spec.Refs.Pulls[0].Number)
-		prev, ok := dupes[n]
-		if !ok {
-			dupes[n] = i
-			continue
-		}
-		cancelIndex := i
-		if (&pjs[prev].Status.StartTime).Before(&pj.Status.StartTime) {
-			cancelIndex = prev
-			dupes[n] = i
-		}
-		toCancel := pjs[cancelIndex]
+	return c.aborter.TerminateOlderPresubmitJobs(pjs, func(toCancel prowapi.ProwJob) error {
 		// Allow aborting presubmit jobs for commits that have been superseded by
 		// newer commits in Github pull requests.
-		if c.config().Plank.AllowCancellations {
+		if c.config().Plank.AllowCancellations || c.config().AllowCancellations {
```

Comment:
Err, the config should be local to a specific controller only

Reply:
I removed the global config flag. 

The idea here it to use both flags for backward compatibility and later to remove the `Plank.AllowCancellations` flag when it is phased out.